modules/showheroes-bsBidAdapter.js

```diff
@@ -1,200 +1,122 @@
-    const isNativeRender = deepAccess(validBidRequests[0], 'renderer');
-    const outstreamOptions = deepAccess(validBidRequests[0], 'params.outstreamOptions');
-    const isBanner = !!validBidRequests[0].mediaTypes.banner || (isOutstream && !(isCustomRender || isNativeRender || isNodeRender));
-    const defaultSchain = validBidRequests[0].schain || {};
-
-    const consentData = bidderRequest.gdprConsent || {};
-    const uspConsent = bidderRequest.uspConsent || '';
-    const gdprConsent = {
-      apiVersion: consentData.apiVersion || 2,
-      gdprApplies: consentData.gdprApplies || 0,
-      consentString: consentData.consentString || '',
+  imp(buildImp, bidRequest, context) {
+    const imp = buildImp(bidRequest, context);
+    const videoContext = deepAccess(bidRequest, 'mediaTypes.video.context');
+    deepSetValue(imp, 'video.ext.context', videoContext);
+    imp.ext = imp.ext || {};
+    imp.ext.params = bidRequest.params;
+    imp.ext.adUnitCode = bidRequest.adUnitCode;
+
+    if (!imp.displaymanager) {
+      imp.displaymanager = 'Prebid.js';
+      imp.displaymanagerver = '$prebid.version$'; // prebid version
     }
 
-    validBidRequests.forEach((bid) => {
-      const videoSizes = getVideoSizes(bid);
-      const bannerSizes = getBannerSizes(bid);
-      const vpaidMode = getBidIdParameter('vpaidMode', bid.params);
-
-      const makeBids = (type, size, isViralize) => {
-        let context = '';
-        let streamType = 2;
-
-        if (type === BANNER) {
-          streamType = 5;
-        } else {
-          context = deepAccess(bid, 'mediaTypes.video.context');
-          if (vpaidMode && context === 'instream') {
-            streamType = 1;
-          }
-          if (context === 'outstream') {
-            streamType = 5;
-          }
-        }
+    if (!isFn(bidRequest.getFloor)) {
+      return imp
+    }
 
-        let rBid = {
-          type: streamType,
-          adUnitCode: bid.adUnitCode,
-          bidId: bid.bidId,
-          context: context,
-          // TODO: fix auctionId leak: https://github.com/prebid/Prebid.js/issues/9781
-          auctionId: bidderRequest.auctionId,
-          start: +new Date(),
-          timeout: 3000,
-          params: bid.params,
-          schain: bid.schain || defaultSchain
-        };
+    let floor = bidRequest.getFloor({
+      currency: 'EUR',
+      mediaType: '*',
+      size: '*',
+    });
+    if (!isNaN(floor?.floor) && floor?.currency === 'EUR') {
+      imp.bidfloor = floor.floor;
+      imp.bidfloorcur = 'EUR';
+    }
+    return imp;
+  },
+  request(buildRequest, imps, bidderRequest, context) {
+    const req = buildRequest(imps, bidderRequest, context);
+    // delete user agent from oRTB, we'll get it from the header
+    (req?.device?.ua) && delete req.device['ua'];
+    // 'sua' is 2.6 standard, we operate with 2.5
+    (req?.device?.sua) && delete req.device['sua'];
```

Comment:
do not necessarily need to check can just call delete (either way is fine just pointing out)

```js
delete req?.device?.ua;
delete req?.device?.sua;
```